test/test_kafka_producer.js

```diff
@@ -142,6 +142,51 @@ test('Kafka producer could write with batched produce.', function testKafkaProdu
     }
 });
 
+test('Kafka producer properly handles large messages.', function testKafkaProducer(assert) {
+    var server = new KafkaRestProxyServer(4444);
+    server.start();
+
+    var PORT = 4444;
+    var configs = {
+        proxyHost: 'localhost',
+        proxyPort: PORT,
+        proxyRefreshTime: 0,
+        batching: true
+    };
+    var producer = new KafkaProducer(configs);
+    producer.connect(onConnect);
+
+    // Specifically test to make sure we batch messages at the 100k
+    // limit correctly
+    var almostTooLargeMessage = new Array(100000 - 7).join('a');
```

Comment:
The threshold is maxBufferSize-8, so is almostTooLargeMessage.length already too large?